tensorflow/stream_executor/dnn.h

```diff
@@ -39,6 +39,11 @@ limitations under the License.
 #include "tensorflow/stream_executor/platform/logging.h"
 #include "tensorflow/stream_executor/platform/port.h"
 
+namespace cudnn_frontend {
```

Comment:
`stream_executor` provides a portable layer, and as you may notice everything in the API isn't target-specific. Since the frontend API is target-specifc Can you move it to `core/kernels`?

Reply:
Yes, I noticed this issue. Not like the `cudnnConvolutionFwdAlgo_t` that we can convert to integers and hide its platform-specific type, the `cudnn_frontend::ExecutionPlan` is a special non-copyable data structure that we have to move it from the backend to the Op's Compute method. I need to think about how to bridge this by using a target agnostic way. Any idea is appreciated.

Reply:
Came up with a way to remove all those deps on cudnn_frontend headers in non cuda_dnn.cc files. I put the change in the new commit. (I will try to split the PR into multiple commits later.)